### PR TITLE
feat(llm): add new haiku into func calling support model

### PR DESCRIPTION
**Pull request by xingyaoww**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [X] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

Add new haiku (`claude-3-5-haiku-20241022`) into func calling support model

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=ghcr.io/all-hands-ai/runtime:1e9dbf3-nikolaik   --name openhands-app-1e9dbf3   ghcr.io/all-hands-ai/runtime:1e9dbf3
```

### CODE REVIEW

**Review by neubig — APPROVED** [review]

**Comment by enyst** [comment]
I just tried the old one today and it didn't work with function calling. 😭 

They've made this one work?

**Comment by xingyaoww** [comment]
@enyst I believe so - should be in feature parity with the new sonnet (they shared some SWE-Bench verified score)! I'm currently running an eval on new haiku (slow down by the rate limit), will merge this PR once we got some decent result

image

**Review by tofarr — APPROVED** [review]

**Comment by enyst** [comment]
The image CI job got stuck for some reason. Restarted.

**Comment by xingyaoww** [comment]
Just realize i added it to prompt caching list but not function calling 😓  add it now!

File: openhands/llm/llm.py [hunk 1]
```diff
@@ -369,7 +371,7 @@ def is_function_calling_active(self) -> bool:
             or self.config.model.split('/')[-1] in FUNCTION_CALLING_SUPPORTED_MODELS
             or any(m in self.config.model for m in FUNCTION_CALLING_SUPPORTED_MODELS)
         )
-        return model_name_supported and (
+        return model_name_supported or (
```
Comment (xingyaoww) [inline openhands/llm/llm.py:374]: We also need to do this (since litellm haven't update their model_info yet)

**Pull request merged**